tests/test_builders/test_build_linkcheck.py

```diff
@@ -1439,3 +1439,87 @@ def test_linkcheck_exclude_documents(app: SphinxTestApp) -> None:
         'uri': 'https://www.sphinx-doc.org/this-is-another-broken-link',
         'info': 'br0ken_link matched br[0-9]ken_link from linkcheck_exclude_documents',
     } in content
+
+
+class CaseSensitiveHandler(BaseHTTPRequestHandler):
```

Comment:
Slightly nitpicky, but: the client and the test case are about case sensitivity -- but this handler is not really.  What it does is to normalise/capitalise the path.  So I'd suggest renaming it:

```suggestion
class CapitalisePathHandler(BaseHTTPRequestHandler):
```

Reply:
Renamed it from `CaseSensitiveHandler` to `CapitalisePathHandler` and updated the docstring to reflect what it actually does.